i18n/checkLocaleKeys.js

```diff
@@ -0,0 +1,90 @@
+import fs from 'fs'
+import path from 'path'
+import { fileURLToPath } from 'url'
+
+// Convert import.meta.url to __dirname equivalent
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
+
+// Define the path to our i18n locales directory
+const localesI18nDirectory = path.join(__dirname, '/locales')
+
+// Import the key-checking function
+const checkLocaleKeys = () => {
+    const localeFilesWithoutEnFileName = fs
+        .readdirSync(localesI18nDirectory)
+        .filter(localeFile => localeFile.endsWith('.json') && localeFile !== 'en.json')
+
+    const enFileName = 'en.json'
+    const enFilePath = path.join(localesI18nDirectory, enFileName)
+    const enFileContent = JSON.parse(fs.readFileSync(enFilePath, 'utf-8'))
+
+    // Function to create a full key path
+    const createFullKey = (prefix, key) => prefix ? `${prefix}.${key}` : key
+
+    // Function to check if value is an object
+    const isNestedObject = value => typeof value === 'object' && value !== null
+
+    // Extract all keys from a locale file
+    const extractAllKeys = (jsonObject, prefix = '') => {
+        const keys = []
+        Object.keys(jsonObject).forEach(key => {
+            const fullKey = createFullKey(prefix, key)
+            if (isNestedObject(jsonObject[key])) {
+                keys.push(...extractAllKeys(jsonObject[key], fullKey))
+            } else {
+                keys.push(fullKey)
+            }
+        })
+        return keys
+    }
+
+    // Function to insert a key-value pair into a nested object
+    const insertMissingKey = (targetObject, keyPath, value) => {
```

Comment:
```suggestion
    const insertMissingKeysAndValues = (valueToSet, keyPath, value) => {
```

This is also setting the values, so we should clarify the name. Also targetObject was a bit confusing